source/_posts/2015-10-11-a-configuration-management-rosetta-stone-configuring-sensu-with-puppet.markdown

```diff
@@ -0,0 +1,352 @@
+
+#### Ansible Sensu Playbook Review
+
+There is no official Sensu Ansible playbook. I was not able to find any playbooks
+support RedHat-based distributions.
+
+I was lucky and was able to use [Mayeu's ansible playbook](https://github.com/Mayeu/ansible-playbook-sensu), in conjuction with
+this [RabbitMQ playbook](https://github.com/Mayeu/ansible-playbook-rabbitmq) on
+my Ubuntu server.
+
+The `sensu_check` module is part of the "Extras", but it is only a very small
+part of deploying Sensu, and it has no cohesion with the playbook that actually
+deploys Sensu itself. There is no way to extend `sensu_check` without forking
+[ansible-modules-extras](http://github.com/ansible/ansible-modules-extras). It
+can't consume arbitrary check metadata.
+
+In the end, to meet my needs I had to construct hashes myself and deploy them
+to disk as JSON. The playbook-provided way to deploy sensu checks is having
+them all contained in the single
+[sensu\_checks](https://github.com/Mayeu/ansible-playbook-sensu#general-variables)
+variable.
+
+### Salt
+
+#### Salt in General
+
+Salt is also a relative new-comer to the configuration management world. As a
+user, Salt feels very similar to Ansible. They both use yaml files to represent
+the desired state of the system. Both use Jinja templates. Both require the
+"advanced" system interaction to happen with the core stuff, and the
+Salt-formulas can be just yaml with no real code.
+
+Salt takes a different approach to sharing community code compared to the
+other configuration management systems. Salt keeps all the official formulas in
+one [GitHub project](https://github.com/saltstack-formulas). The docs recommend
+[forking the formual](https://docs.saltstack.com/en/latest/topics/development/conventions/formulas.html#adding-a-formula-as-a-gitfs-remote)
+for your own use. On the plus side, having "canonical" formulas for common
+tasks reduces duplication and encourages code re-use. The downside is that.. it
+encourages forking? These formulas in general are not that extensive. They
+don't have releases or any kind of testing in place.
+
+Salt's [Pillar](https://docs.saltstack.com/en/latest/topics/pillar/) is a good
+tool. It is similar to Puppet's Hiera. Pro: separating config from code; keep
+the site-specific variables in a separate folder than the formulas. Con:
+formulas have to be "pillar-aware". There is no equivilant to Puppet's
+[Automatic parameter lookup](https://docs.puppetlabs.com/hiera/1/puppet.html#automatic-parameter-lookup).
+
+#### Sensu Salt Formula Review
+
+For my testing, I used the official
+[Salt-formula](https://github.com/saltstack-formulas/sensu-formula). There is
+an sensu-salt repo on the official Sensu project, but it is not really suitable
+for production use in my opinion.
+
+For the most part, the formula did what it said on the tin. Of course, like
+Ansible, the only way I was able to deploy checks in a sane way was to
```

Comment:
I totally get what you mean by 'in a sane way'.  You want other bits of code to be able to independently add Sensu checks.  But did you explain that anywhere?  I can't remember.